packages/cloudflare/src/index.ts

```diff
@@ -490,12 +490,14 @@ export default function createIntegration(args?: Options): AstroIntegration {
 					/**
 					 * These route types are candiates for being part of the `_routes.json` `include` array.
 					 */
+					let notFoundIsSSR = false;
 					const potentialFunctionRouteTypes = ['endpoint', 'page'];
-
 					const functionEndpoints = routes
 						// Certain route types, when their prerender option is set to false, run on the server as function invocations
 						.filter((route) => potentialFunctionRouteTypes.includes(route.type) && !route.prerender)
 						.map((route) => {
+							if (route.component === 'src/pages/404.astro' && route.prerender === false)
```

Comment:
This hardcodes the component property to begin with `"src"`, but that might be different if the user configures [srcDir](https://docs.astro.build/en/reference/configuration-reference/#srcdir).

Reply:
Not blocking thought, as realistically that affects _maybe_ 1 user on the planet.

Reply:
Yeah I won't fix it in this PR. But maybe do another follow-up! :rocket: